custom_components/tahoma/tahoma_device.py

```diff
@@ -109,10 +113,26 @@ def device_info(self) -> Dict[str, Any]:
         manufacturer = self.select_state(CORE_MANUFACTURER_NAME_STATE) or "Somfy"
         model = self.select_state(CORE_MODEL_STATE) or self.device.widget
 
+        name = self.name
+        device_url = self.device_url
+        if "#" in self.device_url:
+            device_url = self.device.deviceurl.split("#", 1)[0]
+            entity_registry = self.hass.data["entity_registry"]
+            name = next(
+                (
+                    entry.original_name
+                    for entity_id, entry in entity_registry.entities.items()
+                    if entry.unique_id == f"{device_url}#1"
+                ),
+                None,
+            )
+        if "battery sensor" in name:
```

Comment:
You can move also this part in TahomaBatterySensor.  Override device_info, call the super, and redifined the name.

Reply:
done

Reply:
done